packages/drivers/local-driver/src/localSessionStorageDb.ts

```diff
@@ -259,13 +259,18 @@ class LocalSessionStorageCollection<T> implements ICollection<T> {
                 }
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 const value = JSON.parse(sessionStorage.getItem(ssKey!)!);
+                let foundMismatch = false;
                 for (const qk of queryKeys) {
                     if (value[qk] !== query[qk]) {
-                        continue;
+                        foundMismatch = true;
```

Comment:
Not sure of this logic but previously it was like keep looping on queryKeys till value[qk] becomes equal to query[qk]. With new logic we just break the loop on first mismatch. Although I am not sure what the query will be but maybe you can explain with an example or something?

Reply:
`continue` and `break` statements here only have the power to manipulate the `for (const qk of queryKeys)` loop, so this loop was literally a no-op.  Regardless of whether the keys match or not, it would finish this loop and then return `value`.  If the `continue` could instead affect the `for (let i = 0; i < sessionStorage.length; i++)` then this logic would have been working as intended.

What it should be doing is see whether the current `ssKey` is a complete match for the query -- if yes (no mismatches) then return it, otherwise (mismatches found) look at the next candidate `ssKey` until we run out or find a match.  So as soon as we find a mismatch we know we can move on to the next `ssKey` candidate.